### ISSUE TITLE
Cannot run aeternity-v5.0.1 on CentOS7

### ISSUE THREAD

**Issue by ost006** [aeternity/aeternity#2955]
On CentOS 7.6.1810, when trying to run, there is a GLIBC version problem error

Please check error logs below

```
=====
===== LOGGING STARTED Tue Oct 29 15:44:40 KST 2019
=====
Exec: /home/ae/aeternity/erts-10.3.5.4/bin/erlexec -boot /home/ae/aeternity/releases/5.0.1/aeternity -mode embedded -boot_var ERTS_LIB_DIR /home/ae/aeternity/lib -config /home/ae/aeternity/releases/5.0.1/sys.config -args_file /home/ae/aeternity/releases/5.0.1/vm.args -- console  --relx-disable-hooks^M
Root: /home/ae/aeternity^M
/home/ae/aeternity^M
Erlang/OTP 21 [erts-10.3.5.4] [source] [64-bit] [smp:1:1] [ds:1:1:64] [async-threads:30] [hipe]^M
^M
2019-10-29 15:44:43.298596 crash_report        #{label=>{proc_lib,crash},report=>[[{initial_call,{supervisor,kernel,['Argument__1']}},{pid,<0.1186.0>},{registered_name,[]},{error_info,{exit,{on_load_function_failed,ecrecover},[{init,run_on_load_handlers,0,[]},{kernel,init,1,[{file,"kernel.erl"},{line,212}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,295}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,374}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,342}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,249}]}]}},{ancestors,[kernel_sup,<0.1159.0>]},{message_queue_len,0},{messages,[]},{links,[<0.1161.0>]},{dictionary,[]},{trap_exit,true},{status,running},{heap_size,376},{stack_size,27},{reductions,265}],[]]}^M
2019-10-29 15:44:43.298652 supervisor_report   #{label=>{supervisor,start_error},report=>[{supervisor,{local,kernel_sup}},{errorContext,start_error},{reason,{on_load_function_failed,ecrecover}},{offender,[{pid,undefined},{id,kernel_safe_sup},{mfargs,{supervisor,start_link,[{local,kernel_safe_sup},kernel,safe]}},{restart_type,permanent},{shutdown,infinity},{child_type,supervisor}]}]}^M
2019-10-29 15:44:43.298152 Error in process ~p on node ~p with exit value:~n~p~n^M
        <0.1187.0>^M
        'aeternity@localhost'^M
        {{badmatch,{error,{load_failed,"Failed to load NIF library: '/lib64/libc.so.6: version `GLIBC_2.18' not found (required by /home/ae/aeternity/lib/ecrecoverprebuilt-274f90ba3d93468e8eec405309b43b6c97ae4246/priv/ecrecover.so)'"}}},[{ecrecover,load,0,[{file,"/home/ae/aeternity_source/_build/default/lib/ecrecover/src/ecrecover.erl"},{line,15}]},{init,'-run_on_load_handlers/2-fun-0-',1,[]}]}^M
2019-10-29 15:44:44.303502 crash_report        #{label=>{proc_lib,crash},report=>[[{initial_call,{application_master,init,['Argument__1','Argument__2','Argument__3','Argument__4']}},{pid,<0.1158.0>},{registered_name,[]},{error_info,{exit,{{shutdown,{failed_to_start_child,kernel_safe_sup,{on_load_function_failed,ecrecover}}},{kernel,start,[normal,[]]}},[{application_master,init,4,[{file,"application_master.erl"},{line,138}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,249}]}]}},{ancestors,[<0.1157.0>]},{message_queue_len,1},{messages,[{'EXIT',<0.1159.0>,normal}]},{links,[<0.1157.0>,<0.1156.0>]},{dictionary,[]},{trap_exit,true},{status,running},{heap_size,610},{stack_size,27},{reductions,193}],[]]}^M
2019-10-29 15:44:44.305373 std_info            #{label=>{application_controller,exit},report=>[{application,kernel},{exited,{{shutdown,{failed_to_start_child,kernel_safe_sup,{on_load_function_failed,ecrecover}}},{kernel,start,[normal,[]]}}},{type,permanent}]}^M
{"Kernel pid terminated",application_controller,"{application_start_failure,kernel,{{shutdown,{failed_to_start_child,kernel_safe_sup,{on_load_function_failed,ecrecover}}},{kernel,start,[normal,[]]}}}"}^M
Kernel pid terminated (application_controller) ({application_start_failure,kernel,{{shutdown,{failed_to_start_child,kernel_safe_sup,{on_load_function_failed,ecrecover}}},{kernel,start,[normal,[]]}}})^M
^M
Crash dump is being written to: erl_crash.dump...done
```

**Comment by tolbrino** [comment]
Did you build the release from source or used a pre-built package?

**Comment by ost006** [comment]
Hi tolbrino.
thank you for your answer

I was built the release from source. (tags/v5.0.1)
(Aeternity-v4.1.0 is available run on CentOS 7)

Please check this issue
or consider providing a pre-build version for CentOS7.

Thank you so much.

**Comment by tolbrino** [comment]
To get up and running you can do the following:

1. Install rust (possibly via rustup)
2. `cd _build/default/lib/ecrecover && make`
3. `cp _build/default/lib/ecrecover/priv/ecrecover.so _build/default/lib/ecrecoverprebuilt/priv_src/unix/linux/`

Then you should be able to boot the system. Note though that you will have to repeat the last 2 steps after you cleaned the local repository.

**Comment by tolbrino** [comment]
@ost006 Did the steps above work for you?

**Comment by tolbrino** [comment]
I've tested my proposed approach, which doesn't work in that simplistic form. This is due to the way the prebuilt binaries are fetched which can't be easily overloaded. Thus I'd consider this to be currently not possible until the build of the binaries is automated. See https://github.com/aeternity/aeternity/issues/3016

Closing this issue since the work which needs to be done is tracked in the referenced issue.
